pycloudflare/services.py

```diff
@@ -1,17 +1,26 @@
 from demands import HTTPServiceClient, HTTPServiceError
+from demands.pagination import (
+    PAGE_PARAM, PAGE_SIZE_PARAM, PAGE_SIZE, PAGINATION_TYPE, RESULTS_KEY,
+    PaginatedResults, PaginationType)
 from six.moves.urllib.parse import urlencode
 
 from pycloudflare.config import get_config
-from pycloudflare.pagination import PaginatedAPIIterator
 
 
 class ZoneNotFound(Exception):
     pass
 
 
-class CloudFlarePageIterator(PaginatedAPIIterator):
-    page_size_param = 'per_page'
-    page_size = 50
+CF_PAGINATION_OPTIONS = {
+    PAGE_SIZE_PARAM: 'per_page',
+    PAGE_SIZE: 50,
+    RESULTS_KEY: None,
+}
+
+
+def cloudflare_paginated_results(fn, args=(), kwargs=None):
```

Comment:
I was matching the signature of PaginatedResults.

It's kind a messy, because we're wrapping a class that wraps a function, that takes arbitrary arguments :)